### ISSUE TITLE
read_gbq() unnecessarily waiting on getting default credentials from Google

### ISSUE THREAD

**Issue by dfontenot** [pydata/pandas-gbq#73]
When attempting to grant pandas access to my GBQ project, I am running into an issue where `read_gbq` is trying to get default credentials, failing / timing out, then printing out a URL to go to to grant the credentials. Since I'm not running this on google cloud platform, I do not expect to be able to get default credentials. In my case, I only want to run the CLI flow (without having oauth call back to my local server).

Here's the code
```
>>> import pandas_gbq as gbq
>>> gbq.read_gbq('SELECT 1', project_id=<project_id>, auth_local_webserver=False)
```

Here's what I see when I trigger a SIGINT once the query is invoked:
```
  File "/usr/lib/python3.5/site-packages/pandas_gbq/gbq.py", line 214, in get_credentials
    credentials = self.get_application_default_credentials()
  File "/usr/lib/python3.5/site-packages/pandas_gbq/gbq.py", line 243, in get_application_default_credentials
    credentials, _ = google.auth.default(scopes=[self.scope])
  File "/usr/lib/python3.5/site-packages/google/auth/_default.py", line 277, in default
    credentials, project_id = checker()
  File "/usr/lib/python3.5/site-packages/google/auth/_default.py", line 274, in <lambda>
    lambda: _get_gce_credentials(request))
  File "/usr/lib/python3.5/site-packages/google/auth/_default.py", line 176, in _get_gce_credentials
    if _metadata.ping(request=request):
  File "/usr/lib/python3.5/site-packages/google/auth/compute_engine/_metadata.py", line 73, in ping
    timeout=timeout)
  File "/usr/lib/python3.5/site-packages/google/auth/transport/_http_client.py", line 103, in __call__
    method, path, body=body, headers=headers, **kwargs)
  File "/usr/lib/python3.5/http/client.py", line 1106, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1151, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/lib/python3.5/http/client.py", line 849, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib/python3.5/socket.py", line 702, in create_connection
    sock.connect(sa)
KeyboardInterrupt
```

I've also tried setting the env variable `GOOGLE_APPLICATIONS_CREDENTIALS` to empty. I'm using pandas-gbq version at commit 64a19b.

**Comment by parthea** [comment]
Looking at the `google-auth` library at https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/compute_engine/_metadata.py#L48, it seems that the default timeout for the request in `_metadata.ping` is 3 seconds unless the environment variable `GCE_METADATA_TIMEOUT` is set to something else. Can you check if the env variable `GCE_METADATA_TIMEOUT` exists in your environment?

**Comment by davidfontenot** [comment]
@parthea I am not seeing `GCE_METADATA_TIMEOUT`. I just timed how long it took for the URL message to show up, and I counted about a minute and 15 seconds. I tried setting `GCE_METADATA_TIMEOUT` to 0, but the call is still waiting on the credentials check.

**Comment by davidfontenot** [comment]
I'm using google-api-python-client-1.6.2, which is bringing in google-auth-1.0.1

**Comment by parthea** [comment]
So for some reason the default timeout of 3 seconds in `google-auth` is not used. You'll likely see similar behaviour if you run the following python code.

```
from google.auth.compute_engine import _metadata
import google.auth.transport._http_client
request = google.auth.transport._http_client.Request()
_metadata.ping(request=request)
```

Can you try the above code and opening a new issue in the `google-auth` project here if you don't get a result in 3 seconds or less?
https://github.com/GoogleCloudPlatform/google-auth-library-python/issues/new

**Comment by davidfontenot** [comment]
@parthea confirmed that the above code snippet takes longer than 3 seconds to run. Thanks for digging into this issue. I will open a ticket with them and link it over here.

As for this library, would you consider adding a more explicit way to opt out of trying to get the default credentials?

**Comment by parthea** [comment]
Thanks @davidfontenot . Please feel free to reference this issue from the one you create in `google-auth`. I'd like to spend some time to dig into the root cause before adding a workaround. Let's keep this issue open until we have a solution. 

cc @tswast

**Comment by parthea** [comment]
Thanks @davidfontenot ! This issue was caused by bug in the `google-auth` library which is now fixed. You'll need to install the `google-auth` library from source until the next release of `google-auth`. I'm going to close this issue but feel free to re-open if you're still having trouble.

**Comment by jreback** [comment]
this needs to remain open until the new version is released and tested

**Comment by parthea** [comment]
No problem. I have a mac at home that I can use to test the fix unless @davidfontenot tries it first.

**Comment by davidfontenot** [comment]
@parthea @jreback I've updated my requirements.txt to point to google-auth's latest commit. If I do not set `GCE_METADATA_TIMEOUT=0` it takes about 6 seconds to present a url. If I do set it it takes about 2 seconds. Not sure if there are some other network calls going on during this time. Also not seeing any effect by setting `GOOGLE_APPLICATIONS_CREDENTIALS=`.

**Comment by parthea** [comment]
> I've updated my requirements.txt to point to google-auth's latest commit. If I do not set GCE_METADATA_TIMEOUT=0 it takes about 6 seconds to present a url. If I do set it it takes about 2 seconds. Not sure if there are some other network calls going on during this time. 

It sounds like the issue is no longer present with the latest `google-auth` library. Can you confirm its ok to close this issue?

>Also not seeing any effect by setting GOOGLE_APPLICATIONS_CREDENTIALS=.

`google.auth.default()` will still check for credentials regardless if `GOOGLE_APPLICATIONS_CREDENTIALS` is set. There are a few places where `google-auth` checks for credentials. One check is whether the machine is running in a GCE environment.
https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/_default.py#L197

**Comment by davidfontenot** [comment]
@parthea yes, the issue is no longer present. You may close this issue. Thanks again for your help!

**Comment by parthea** [comment]
I'm glad the issue is resolved! 

@jreback Is it ok to close this based on @davidfontenot 's test or would you prefer to keep this open until the next release of `google-auth`?

**Comment by jreback** [comment]
as I said above. this prob needs a min version on ``google-auth`` for the library. So would need to wait until that is out to have a tests / config / whatsnew note for this.

**Comment by maxim-lian** [comment]
Closing as upstream issue ref https://github.com/pydata/pandas-gbq/pull/143 

Please reopen if not upstream issue
